docs/sdk/src/guides/your_first_node.rs

```diff
@@ -1 +1,311 @@
+				.arg("--release")
+				.arg("-p")
+				.arg("staging-chain-spec-builder")
+				.assert()
+				.success();
+		}
+		assert!(find_release_binary(CHAIN_SPEC_BUILDER).is_some());
+	}
+
+	fn maybe_build_omni_node() {
+		if find_release_binary(OMNI_NODE).is_none() {
+			println!("Building polkadot-omni-node...");
+			Command::new("cargo")
+				.arg("build")
+				.arg("--release")
+				.arg("-p")
+				.arg("polkadot-omni-node")
+				.assert()
+				.success();
+		}
+	}
+
+	fn test_runtime_preset(runtime: &'static str, block_time: u64, maybe_preset: Option<PresetId>) {
+		sp_tracing::try_init_simple();
+		maybe_build_runtimes();
+		maybe_build_chain_spec_builder();
+		maybe_build_omni_node();
+
+		let chain_spec_builder =
+			find_release_binary(&CHAIN_SPEC_BUILDER).expect("we built it above; qed");
+		let omni_node = find_release_binary(OMNI_NODE).expect("we built it above; qed");
+		let runtime_path = find_wasm(runtime).expect("we built it above; qed");
+
+		let random_seed: u32 = rand::thread_rng().gen();
+		let chain_spec_file = std::env::current_dir()
+			.unwrap()
+			.join(format!("{}_{}_{}.json", runtime, block_time, random_seed));
+
+		Command::new(chain_spec_builder)
+			.args(["-c", chain_spec_file.to_str().unwrap()])
+			.arg("create")
+			.args(&["--para-id", "1000", "--relay-chain", "dontcare"])
+			.args(["-r", runtime_path.to_str().unwrap()])
+			.args(match maybe_preset {
+				Some(preset) => vec!["named-preset".to_string(), preset.to_string()],
+				None => vec!["default".to_string()],
+			})
+			.assert()
+			.success();
+
+		let output = Command::new(omni_node)
+			.arg("--tmp")
+			.args(["--chain", chain_spec_file.to_str().unwrap()])
+			.args(["--dev-block-time", block_time.to_string().as_str()])
+			.timeout(std::time::Duration::from_secs(10))
+			.output()
+			.unwrap();
+
+		std::fs::remove_file(chain_spec_file).unwrap();
+
+		// uncomment for debugging.
+		// println!("output: {:?}", output);
+
+		let expected_blocks = (10_000 / block_time).saturating_div(2);
+		assert!(expected_blocks > 0, "test configuration is bad, should give it more time");
+		assert!(String::from_utf8(output.stderr)
+			.unwrap()
+			.contains(format!("Imported #{}", expected_blocks).to_string().as_str()));
+	}
+
+	#[test]
+	fn works_with_different_block_times() {
+		test_runtime_preset(PARA_RUNTIME, 100, Some(DEV_RUNTIME_PRESET.into()));
+		test_runtime_preset(PARA_RUNTIME, 3000, Some(DEV_RUNTIME_PRESET.into()));
+
+		// we need this snippet just for docs
+		#[docify::export_content(csb)]
+		fn build_para_chain_spec_works() {
+			let chain_spec_builder = find_release_binary(&CHAIN_SPEC_BUILDER).unwrap();
+			let runtime_path = find_wasm(PARA_RUNTIME).unwrap();
+			let output = "/tmp/demo-chain-spec.json";
+			Command::new(chain_spec_builder)
+				.args(["-c", output])
+				.arg("create")
+				.args(&["--para-id", "1000", "--relay-chain", "dontcare"])
+				.args(["-r", runtime_path.to_str().unwrap()])
+				.args(vec!["named-preset", "development"])
```

Comment:
`vec!` could be removed?
```suggestion
				.args(&["named-preset", "development"])
```

Reply:
damn it seems like I used all possible combinations of `[]`, `&[]` and `vec![]` :))